ClientApp/src/Servises/BookService.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Linq;
+using dreamHotel.Models;
+using dreamHotel.Controllers;
+using dreamHotel.Data;
+using Microsoft.AspNetCore.Mvc;
+
+namespace dreamHotel.Services
+{
+    public class BookService
+    {
+        private ApplicationDbContext _context;
+
+        public BookService(ApplicationDbContext context)
+        {
+            this._context = context;
+        }
+
+        public Reservation CreateСompleteReservation(NewReservation newReservation)
+        {
+            decimal OneDayPrice = newReservation.Room.Price;
+            var countStayDay = newReservation.CheckOutDate.Subtract(newReservation.CheckInDate).Days;
+            decimal TotalCost = countStayDay * OneDayPrice;
+
+            return new Reservation(
+                newReservation.Id,
+                newReservation.UserId,
+                newReservation.Room,
+                newReservation.CheckInDate,
+                newReservation.CheckOutDate,
+                OneDayPrice,
+                TotalCost,
+                false,
+                newReservation.Breakfast
+                );
+        }
+
+        public Reservation Create(NewReservation newReservation)
+        {
+            var completeReservation = CreateСompleteReservation(newReservation);
+            _context.Reservations.Add(completeReservation);
+            _context.SaveChanges();
+
+            return completeReservation;
+        }
+
+        public IEnumerable<Reservation> AllReservation(int userId)
+        {
+            return _context.Reservations.Where(reservation => reservation.UserId == userId);
+
+        }
+
+        public Reservation Update(int id, Reservation newReservation)
+        {
+            foreach (var Reservation in _context.Reservations)
```

Comment:
This is incorrect solution. You are fetching and iterating through all rows in database. Instead we have ``DbSet.Find(int primaryKey)`` function, which works like it was supposed and returns ``null`` if not found.
```suggestion
            var reservation = _context.Reservations.Find(id);
            if (reservation != null)
            {
                // code...
            }
```